clients/src/main/java/org/apache/kafka/common/record/DefaultRecord.java

```diff
@@ -369,6 +369,71 @@ private static DefaultRecord readFrom(ByteBuffer buffer,
         }
     }
 
+    /**
+     * Compute size in bytes of record body and size in bytes of a full record without depressed.
+     *
+     * @param input
+     * @param baseOffset
+     * @param baseTimestamp
+     * @param baseSequence
+     * @param logAppendTime
+     * @return
+     * @throws IOException
+     */
+    public static SimplifiedDefaultRecord simplifiedreadFrom(DataInput input,
+                                                             long baseOffset,
+                                                             long baseTimestamp,
+                                                             int baseSequence,
+                                                             Long logAppendTime) throws IOException {
+        int sizeOfBodyInBytes = ByteUtils.readVarint(input);
+        int totalSizeInBytes = ByteUtils.sizeOfVarint(sizeOfBodyInBytes) + sizeOfBodyInBytes;
```

Comment:
@ijuma  We removed code "**input.readFully(recordBuffer.array(), 0, sizeOfBodyInBytes)**" at  function **simplifiedreadFrom**() ,therefor  we will not call function **readBlock**() at class **KafkaLZ4BlockInputStream**.So we don't decompress buffered data.